File: Programmers/고득점 kit/스택, 큐/기능개발.py
import math

def solution(progresses: list, speeds: list) -> list:
    answer: list = []
    work_day: list = [math.ceil((100 - a) / b) for a, b in zip(progresses, speeds)]
    prev: int = work_day[0]
    count: int = 0

    print(work_day)

    for i in work_day:
        if prev >= i:
            count += 1
        else:
            answer.append(count)
            count, prev = 1, i

    answer.append(count)
    return answer

progress: list = [93, 30, 55]
speeds: list = [1, 30, 5]

print(solution(progress, speeds))

File: Programmers/고득점 kit/스택, 큐/기능개발.js
function solution(progresses, speeds) {
    const answer = [];
    const workDays = progresses.map((progress, index) => Math.ceil((100 - progress) / speeds[index]));
    let prev = workDays[0];
    let count = 0;

    workDays.forEach((workDay) => {
        if (prev >= workDay) {
            count += 1;
            return;
        }
        if (prev < workDay) {
            answer.push(count);
            count = 1;
            prev = workDay;
            return;
        }   
    });
    answer.push(count);
    return answer;
}

console.log(solution([93, 30, 55], [1, 30, 5]) === [2, 1])